scripts/replay-tx.js

```diff
@@ -7,25 +12,43 @@ const { getContractInterface } = require("@eth-optimism/contracts");
 const { JsonRpcProvider, getDefaultProvider } = require("@ethersproject/providers");
 const program = new Command();
 
-program.requiredOption("-h, --hash <number>", "specify transaction hash");
 program.requiredOption("-n, --network <string>", "specify network");
-program.option("-k, --key <number>", "specify private key");
+program.requiredOption("-k, --key <number>", "specify private key");
+program.option("-h, --hash <number>", "specify transaction hash");
+program.option("-b, --blockNumber <number>", "specify a block number");
+
 program.parse(process.argv);
 const argOptions = program.opts();
 
 const mainnetProvider = getDefaultProvider(argOptions.network);
 
-const L1_MESSENGER_PROXY = "0xD1EC7d40CCd01EB7A305b94cBa8AB6D17f6a9eFE";
+// Addresses from May 11 regenesis
+const MAINNET_L1_MESSENGER_PROXY = "0x902e5fF5A99C4eC1C21bbab089fdabE32EF0A5DF";
+const KOVAN_L1_MESSENGER_PROXY = "0x78b88FD62FBdBf67b9C5C6528CF84E9d30BB28e0";
 
 const wallet = new Wallet(argOptions.key, mainnetProvider);
 const proxyL1Messenger = new Contract(L1_MESSENGER_PROXY, getContractInterface("OVM_L1CrossDomainMessenger"), wallet);
 
 const main = async () => {
-  const receipt = await mainnetProvider.getTransactionReceipt(argOptions.hash);
+  if (!argOptions.hash && !argOptions.blockNumber) {
+    console.error("Must provide a hash or block number.");
+    return;
+  }
+
+  if (argOptions.hash) {
+    replayMessage(argOptions.hash);
+  } else {
+    // TODO: loop over blocks starting at argOptions.blockNumber, get all failed transactions and replay them
```

Comment:
🤷‍♂️

Reply:
Are you planning on adding this in this PR?

Reply:
yep! Liam & Kevin unblocked me. thanks